drop-core/src/status.rs

```diff
@@ -0,0 +1,35 @@
+#[derive(Debug, Clone, Copy)]
+#[repr(u32)]
+pub enum Status {
+    Canceled = 1,
+    BadPath = 2,
+    BadFile = 3,
+    ServiceStop = 6,
+    BadTransfer = 7,
+    BadTransferState = 8,
+    BadFileId = 9,
+    Io = 15,
+    DirectoryNotExpected = 17,
+    TransferLimitsExceeded = 20,
+    MismatchedSize = 21,
+    UnexpectedData = 22,
+    InvalidArgument = 23,
+    TransferTimeout = 24,
+    WsServer = 25,
+    WsClient = 26,
+    AddrInUse = 27,
+    FileModified = 28,
+    FilenameTooLong = 29,
+    AuthenticationFailed = 30,
+    StorageError = 31,
+    DbLost = 32,
+}
+
+impl serde::Serialize for Status {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        (*self as u32).serialize(serializer)
+    }
+}
```

Comment:
Can't we decorate the enum with `#[derive(Serialize)]` ?

Reply:
Um, yes. But it will be serialized as a string representation for example `DbLost` and not `32`.
TBH I would prefer the string variant but I was trying to be in line with the other events where the integer is used

Reply:
Thanks for explaininig.